### ISSUE TITLE
Synchronous updates don't work on $slots

### ISSUE THREAD

**Issue by matt-oconnell** [vuejs/vue-test-utils#66]
**Version**

1.0.0-beta

**Steps to reproduce**

Create a component that uses slots:
```vue
// SlotComponent.vue
<template>
  <div>
    <slot></slot>
  </div>
</template>
```

Use this component in some Parent.vue:
```vue
<template>
  <SlotComponent>
    <div :class="[{ 'is-on': on }, 'inner']"></div>
  </SlotComponent>
</template>

<script>
//...
export default {
  //...
  props: {
    on: {
      default: false,
      type: Boolean
    }
  },
  // ...
}
</script>
```

**What is expected?**

Setting props should synchronously update the template, but this test fails:
```js
const parentWrapper = mount(Parent)
const innerEl = parentWrapper.find('.inner')

expect(innerEl.hasClass('is-on')).to.equal(false)

parentWrapper.setProps({
  on: true
})

expect(innerEl.hasClass('is-on')).to.equal(true)
```

**What is actually happening?**
`nextTick` is still required to update slot content.

**Solution**
Looks like Vue's `vm._update` function clones existing slots and does not update them unless the parent has updated. To fix this in our test we can trigger an update on our `SlotComponent.vue`:

```js
parentWrapper.setProps({
  on: true
}) // calls parentWrapper.update

childWrapper = parentWrapper.find(SlotComponent)
childWrapper.update() // no nextTick required with this
```

**Comment by eddyerburgh** [comment]
I've fixed this in b714ce3

Update now loops through child Vue instances and calls update on them

**Comment by matt-oconnell** [comment]
Amazing! Glad I could help and thanks for the quick response. I tested this fix and it seems to work for deeply nested slot content as well.
